internal/integration/api/apply-config.go

```diff
@@ -135,62 +138,67 @@ func (suite *ApplyConfigSuite) TestApply() {
-		cfg.MachineConfig.MachineSysctls = make(map[string]string)
-	}
+		if cfg.MachineConfig.MachineSysctls == nil {
+			cfg.MachineConfig.MachineSysctls = make(map[string]string)
+		}
 
-	cfg.MachineConfig.MachineSysctls[applyConfigNoRebootTestSysctl] = applyConfigNoRebootTestSysctlVal
+		cfg.MachineConfig.MachineSysctls[applyConfigNoRebootTestSysctl] = applyConfigNoRebootTestSysctlVal
 
-	cfgDataOut, err := cfg.Bytes()
-	suite.Require().NoError(err, "failed to marshal updated machine config data (node %q)", node)
+		cfgDataOut, err := cfg.Bytes()
+		suite.Require().NoError(err, "failed to marshal updated machine config data (node %q)", node)
 
-	_, err = suite.Client.ApplyConfiguration(nodeCtx, &machineapi.ApplyConfigurationRequest{
-		OnReboot: true,
-		Data:     cfgDataOut,
-	})
-	suite.Require().NoError(err, "failed to apply deferred configuration (node %q): %w", node)
+		_, err = suite.Client.ApplyConfiguration(nodeCtx, &machineapi.ApplyConfigurationRequest{
+			Data: cfgDataOut,
+			Mode: mode,
+		})
+		suite.Require().NoError(err, "failed to apply deferred configuration (node %q): %w", node)
 
-	// Verify configuration change
-	var newProvider config.Provider
+		// Verify configuration change
+		var newProvider config.Provider
 
-	newProvider, err = suite.ReadConfigFromNode(nodeCtx)
+		newProvider, err = suite.ReadConfigFromNode(nodeCtx)
 
-	suite.Require().NoError(err, "failed to read updated configuration from node %q: %w", node)
+		suite.Require().NoError(err, "failed to read updated configuration from node %q: %w", node)
 
-	suite.Assert().Equal(
-		newProvider.Machine().Sysctls()[applyConfigNoRebootTestSysctl],
-		applyConfigNoRebootTestSysctlVal,
-	)
+		suite.Assert().Equal(
+			newProvider.Machine().Sysctls()[applyConfigNoRebootTestSysctl],
+			applyConfigNoRebootTestSysctlVal,
+		)
 
-	cfg, ok = newProvider.Raw().(*v1alpha1.Config)
-	suite.Require().True(ok)
+		cfg, ok = newProvider.Raw().(*v1alpha1.Config)
+		suite.Require().True(ok)
 
-	// revert back
-	delete(cfg.MachineConfig.MachineSysctls, applyConfigNoRebootTestSysctl)
+		// revert back
+		delete(cfg.MachineConfig.MachineSysctls, applyConfigNoRebootTestSysctl)
 
-	cfgDataOut, err = cfg.Bytes()
-	suite.Require().NoError(err, "failed to marshal updated machine config data (node %q)", node)
+		cfgDataOut, err = cfg.Bytes()
+		suite.Require().NoError(err, "failed to marshal updated machine config data (node %q)", node)
 
-	_, err = suite.Client.ApplyConfiguration(nodeCtx, &machineapi.ApplyConfigurationRequest{
-		OnReboot: true,
-		Data:     cfgDataOut,
-	})
-	suite.Require().NoError(err, "failed to apply deferred configuration (node %q): %w", node)
+		_, err = suite.Client.ApplyConfiguration(nodeCtx, &machineapi.ApplyConfigurationRequest{
+			Data: cfgDataOut,
+			Mode: mode,
+		})
+		suite.Require().NoError(err, "failed to apply deferred configuration (node %q): %w", node)
+	}
 }
 
 // TestApplyConfigRotateEncryptionSecrets verify key rotation by sequential apply config calls.
```

Comment:
should we update this function as well?

Reply:
Do you mean that we should add explicit `mode` here? Auto mode should pick reboot anyways.

Reply:
ah yeah, you're right... it wasn't immediate